modules/kmm-building-in-cluster.adoc

```diff
@@ -0,0 +1,47 @@
+// Module included in the following assemblies:
+//
+// * hardware_enablement/kmm-kernel-module-management.adoc
+
+:_content-type: CONCEPT
+[id="kmm-building-in-cluster_{context}"]
+
+= Building in the cluster
+
+KMM can build module loader images in the cluster. Follow these guidelines:
+
+* Provide build instructions using the `build` section of a kernel mapping.
+* Copy the `Dockerfile` for your container image into a `ConfigMap` resource, under the `dockerfile` key.
+* Ensure that the `ConfigMap` is located in the same namespace as the `Module`.
+
+KMM checks if the image name specified in the `containerImage` field exists. If it does, the build is skipped.
+
+Otherwise, KMM creates a `Build` resource to build your image. After the image is built, KMM proceeds with the `Module` reconciliation. See the following example.
+
+[source,yaml]
+----
+# ...
+- regexp: '^.+$'
+  containerImage: "some.registry/org/<my_kmod>:${KERNEL_FULL_VERSION}"
+  build:
+    buildArgs:  <1>
+      - name: ARG_NAME
+        value: <some_value>
+    secrets: <2>
+      - name: <some_kubernetes_secret> <3>
+    baseImageRegistryTLS:
+      insecure: false <4>
+      insecureSkipTLSVerify: false <5>
+    dockerfileConfigMap:  <6>
+      name: <my_kmod_dockerfile>
+  registryTLS:
+    insecure: false <7>
+    insecureSkipTLSVerify: false <8>
+----
+<1> Optional.
+<2> Optional.
+<3> Will be mounted in the build pod as `/run/secrets/some-kubernetes-secret`.
+<4> Optional. Avoid using this parameter. If set to `true`, the build will be allowed to pull the image in the Dockerfile `FROM` instruction using plain HTTP.
```

Comment:
We've been doing _Optional:_ rather than _Optional._ in OCP docs.

Reply:
I will change these to _Optional:_ in the cases when a sentence follows.